src/plugins/vis_type_table_new/public/components/table_vis_component.tsx

```diff
@@ -51,8 +52,12 @@ export const TableVisComponent = ({
       const colIndex = columns.findIndex((col) => col.id === columnId);
       const column = columns[colIndex];
       // use formatter to format raw content
-      // this can format date and percentage data
-      const formattedContent = column.formatter.convert(rawContent, 'text');
+      // this can format url, date and percentage data
+      const htmlContent = column.formatter.convert(rawContent, 'html');
+      const formattedContent = (
+        // eslint-disable-next-line
+        <div dangerouslySetInnerHTML={{ __html: dompurify.sanitize(htmlContent) }} />
```

Comment:
Is this thei only way to do this? afaik the goal here was just to make links possible for table cell elements. Cant we do that in a simpler way without allowing full blown HTML in a table cell?

Reply:
Hi @ashwin-pc, there are 3 types for url format that could be set into indexPattern field. There are link, img and audio. 
There are two converter in OSD, one is html and one is text. For these 3 types, here is a compare of text converter vs html converter

- img
```
"/plugins/indexPatternManagement/assets/icons/177.120.218.48.png"
vs
"<span ng-non-bindable><img src="/plugins/indexPatternManagement/assets/icons/177.120.218.48.png" alt="A dynamically-specified image located at /plugins/indexPatternManagement/assets/icons/177.120.218.48.png" style="width:auto; height:auto; max-width:none; max-height:none;"></span>"
```

- audio 
```
"177.120.218.48"
Vs
"<span ng-non-bindable><audio controls preload="none" src="177.120.218.48"></span>"
```
- link
```
"177.120.218.48"
vs
"<span ng-non-bindable><a href="http://localhost:5603/jxv/app/177.120.218.48" target="_blank" rel="noopener noreferrer">177.120.218.48</a></span>"
```

@AMoo-Miki suggested using `EUILink`, but it only works form link. It doesn't work for img and audio. To make it format correctly, we have to use html converter then we need to translate these html contents.

Reply:
I opened an issue [here](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2932)

Reply:
Thanks for the detailed context @ananzh. Yeah I can get behind that solution.